by-language/csharp-npgsql/DemoTypes.cs

```diff
@@ -336,6 +352,102 @@ public async Task<List<BasicPoco>> ArrayPocoExample()
+                @geoshape
+            );
+            """, conn))
+            {
+                var point = new Point(new Position(85.43, 66.23));
+                cmd.Parameters.AddWithValue("geoshape", NpgsqlDbType.Geometry, point);
+                cmd.ExecuteNonQuery();
+            }
+
+            // Flush data.
+            await RefreshTable();
+        }
+
+        public async Task InsertGeoJsonString()
+        {
+            /***
+             * Communicate GeoJSON types as strings, marshall from/to GeoJSON types manually.
+             */
+            Console.WriteLine("Running InsertGeoRaw");
+
+            // Insert single data point.
+            await using (var cmd = new NpgsqlCommand("""
+            INSERT INTO testdrive.example (
+                "geoshape"
+            ) VALUES (
+                @geoshape
+            );
+            """, conn))
+            {
+                var point = new Point(new Position(85.43, 66.23));
+                var poly = new Polygon([
+                    new LineString([
+                        new Position(longitude: 5.0, latitude: 5.0),
+                        new Position(longitude: 5.0, latitude: 10.0),
+                        new Position(longitude: 10.0, latitude: 10.0),
+                        new Position(longitude: 10.0, latitude: 5.0),
+                        new Position(longitude: 5.0, latitude: 5.0),
+                    ])
+                ]);
+                // TODO: Can GEO_SHAPE types be directly marshalled to a .NET GeoJSON type?
+                //       Currently, `InsertGeoJsonTyped` does not work yet.
+                cmd.Parameters.AddWithValue("geoshape", NpgsqlDbType.Json, JsonConvert.SerializeObject(point));
+                cmd.ExecuteNonQuery();
+
+                cmd.Parameters.Clear();
+
+                cmd.Parameters.AddWithValue("geoshape", NpgsqlDbType.Json, JsonConvert.SerializeObject(poly));
+                cmd.ExecuteNonQuery();
+            }
+
+            // Flush data.
+            await RefreshTable();
+
+        }
+
+        public async Task<Point?> GeoJsonTypesExample()
+        {
+            Console.WriteLine("Running GeoJsonTypesExample");
+
+            // Provision data.
+            await CreateTable();
+            // await InsertGeoJsonTyped();
+            await InsertGeoJsonString();
+
+            // Query back data.
+            await using (var cmd = new NpgsqlCommand("SELECT * FROM testdrive.example", conn))
+            await using (var reader = cmd.ExecuteReader())
+            {
+                reader.Read();
+                // TODO: Can GEO_SHAPE types be directly marshalled to a .NET GeoJSON type?
+                //       Currently, `InsertGeoJsonTyped` does not work yet.
+                var obj = reader.GetFieldValue<JsonDocument>("geoshape");
+                var geoJsonObject = JsonConvert.DeserializeObject<Point>(obj.RootElement.ToString());
+                return (Point?) geoJsonObject;
```

Comment:
Reading data is not directly related to how data is ingested. Shouldn't the marshalling work when enabling GeoJSON by `dataSourceBuilder.UseGeoJson();`? Afaik the output JSON should be a valid GeoJSON format, at least for simple structures like Points.

Reply:
Unfortunately, the marshalling does not work, see conversation [here](https://github.com/crate/cratedb-examples/pull/782#pullrequestreview-2518693533).

Reply:
@kneth suggested to track this into the future by using a dedicated ticket. Thanks.

> Can GEO_SHAPE types be directly marshalled to a .NET GeoJSON type?
> Currently, `InsertGeoJsonTyped` does not work yet.

@coderabbitai: Please pick up some context from relevant conversations here, and converge that into a dedicated issue for us. Thanks.